src/models/transactions/paymentTransaction.ts

```diff
@@ -0,0 +1,113 @@
+    DestinationTag?: number
+    InvoiceID?: string
+    Paths?: Path[]
+    SendMax?: Amount
+    DeliverMin?: Amount
+    Flags?: number | PaymentTransactionFlags
+}
+
+/**
+ * @param {PaymentTransaction} tx A Payment Transaction.
+ * @returns {void}
+ * @throws {ValidationError} When the PaymentTransaction is malformed.
+ */
+export function verifyPaymentTransaction(tx: PaymentTransaction): void {
+    verifyBaseTransaction(tx)
+    
+    if (tx.Amount === undefined) {
+        throw new ValidationError('PaymentTransaction: missing field Amount')
+    }
+    
+    if (!isAmount(tx.Amount)) {
+        throw new ValidationError('PaymentTransaction: invalid Amount')
+    }
+    
+    if (tx.Destination === undefined) {
+        throw new ValidationError('PaymentTransaction: missing field Destination')
+    }
+
+    if (!isAmount(tx.Destination)) {
+        throw new ValidationError('PaymentTransaction: invalid Destination')
+    }
+
+    if (tx.DestinationTag !== undefined && typeof tx.DestinationTag !== 'number') {
+        throw new ValidationError('PaymentTransaction: DestinationTag must be a number')
+    }
+    
+    if (tx.InvoiceID !== undefined && typeof tx.InvoiceID !== 'string') {
+        throw new ValidationError('PaymentTransaction: InvoiceID must be a string')
+    }
+    
+    if (tx.Paths !== undefined && !isPaths(tx.Paths)) {
+        throw new ValidationError('PaymentTransaction: invalid Paths')
+    }
+    
+    if (tx.SendMax !== undefined && !isAmount(tx.SendMax)) {
+        throw new ValidationError('PaymentTransaction: invalid SendMax')
+    }
+    
+    if (tx.DeliverMin !== undefined) {
+        const isTfPartialPayment = typeof tx.Flags === 'number' ?
+            isFlagEnabled(tx.Flags, PaymentTransactionFlagsEnum.tfPartialPayment) :
+            tx.Flags?.tfPartialPayment ?? false
+
+        if (!isTfPartialPayment) {
+            throw new ValidationError('PaymentTransaction: tfPartialPayment flag required with DeliverMin')
+        }
+
+        if (!isAmount(tx.DeliverMin)) {
+            throw new ValidationError('PaymentTransaction: invalid DeliverMin')
+        }
+    }
+}
+
+function isPaths(paths: Path[]): boolean {
+    if (!Array.isArray(paths) || paths.length === 0) {
+        return false
+    }
+    
+    for (const i in paths) {
+        const path = paths[i]
+        if (!Array.isArray(path) || path.length === 0) {
+            return false
+        }
+        
+        for (const j in path) {
+            const pathStep = path[j]
+            const { account, currency, issuer } = pathStep
+            if (
+                (account !== undefined && typeof account !== 'string') ||
+                (currency !== undefined && typeof currency !== 'string') ||
+                (issuer !== undefined && typeof issuer !== 'string')
+            ) {
+                return false
+            }
+        }
+    }
+
+    return true;
+}
+
+function isFlagEnabled(Flags: number, checkFlag: number): boolean {
```

Comment:
This would be an awesome function to put in `utils`. 

You can do it in this PR if you'd like, or we can do that when we're working on sign+submit flow.

Reply:
I agree! I moved it to utils and added tests for it 🤠